assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -0,0 +1,391 @@
+  const mapContainerBounds = new Bounds([0, 0], [width, height])
+
+  // ```
+  // vpcElement.getBoundingClientRect().right - mapElement.getBoundingClientRect().left
+  //  -> 445
+  // ```
+  // Create a new inner bounds from the map bounds + "padding" to shrink the
+  // inner bounds
+  // In this case, we get the top left of the inner bounds by padding the left
+  // with the distance from the right side of the VPC to the left side of the
+  // map container
+  const topLeft = new Point(445, 0)
+  const innerBounds = new Bounds(topLeft, mapContainerBounds.getBottomRight())
+  // The "new center" is the offset between the two bounding boxes centers
+  const offset = innerBounds
+    .getCenter()
+    .subtract(mapContainerBounds.getCenter())
+
+  const targetZoom = 16
+  const targetPoint = map
+      // Project the target point into screenspace for the target zoom
+      .project(points[0], targetZoom)
+      // Offset the target point in screenspace to move the center of the map
+      // to apply the padding to the center
+      .subtract(offset),
+    // convert the target point to worldspace from screenspace
+    targetLatLng = map.unproject(targetPoint, targetZoom)
+
+  // Zoom/Pan center of map to offset location in worldspace
+  map.setView(targetLatLng, targetZoom)
+}
+
+const useFollowingStateWithSelectionLogic = (
+  selectedVehicleId: string | null,
+  selectedVehicleRef: VehicleOrGhost | null
+) => {
+  const state = useInteractiveFollowerState(),
+    { setShouldFollow } = state
+
+  // when the selected vehicle ID and last api do or don't reference the same
+  // vehicle
+  const vehicleIdDiffers = selectedVehicleId !== selectedVehicleRef?.id
+  useEffect(() => {
+    // Only update the shouldFollow state once the cached value agrees with
+    // the selection state.
+    // Otherwise the follower may try to center on stale data from the
+    // previous selection before `useVehicleForId` resolves to it's next value.
+    if (selectedVehicleId !== null && vehicleIdDiffers === false) {
+      setShouldFollow(true)
+    }
+  }, [vehicleIdDiffers, setShouldFollow, selectedVehicleId])
+  return state
+}
+
+const selectedEntityRoutePattern = (
+  selectedEntity: LiveSelectedEntity | null
+): { routeId: RouteId; routePatternId: RoutePatternId | null } | null => {
+  switch (selectedEntity?.type) {
+    case SelectedEntityType.Route:
+      return {
+        routeId: selectedEntity.routeId,
+        routePatternId: selectedEntity.routePatternId,
+      }
+    case SelectedEntityType.Vehicle:
+      return selectedEntity.vehicleOrGhost?.routeId
+        ? {
+            routeId: selectedEntity.vehicleOrGhost.routeId,
+            routePatternId: selectedEntity.vehicleOrGhost.routePatternId,
+          }
+        : null
+    default:
+      return null
+  }
+}
+type LiveSelectedEntity =
+  | SelectedRoute
+  | { type: SelectedEntityType.Vehicle; vehicleOrGhost: VehicleOrGhost | null }
+  | null
```

Comment:
**suggestion:** I got a little confused between this type and `SelectedVehicle` in `searchPageState.ts`, since here the value contains an actual vehicle or ghost but there it just has an ID. Similarly for `SelectedRoute` the payload is actually a combination of IDs, not a route data structure. Maybe calling the types in `searchPageState.ts` something like `SelectedVehicleId` and so on would make it a little bit clearer?

Reply:
I tried out the suggestion of `SelectedVehicleId` - let me know what you think!